### ISSUE TITLE
Training issue on Person Detection

### ISSUE THREAD

**Issue by omair18** [opencv/training_toolbox_caffe#17]
I am trying to finetune person-detection network using pretrained weights ```person_detection_0022.caffemodel``` and my data looks like this 

``` <annotations count="9829">
    <track id="0" label="person">
      <box frame="0" occluded="0" xbr="580" xtl="463" ybr="534" ytl="258">
        <attribute name="action">action_name</attribute>
      </box>
    </track>
``` 
Loss at 1st iteration and after 150000 iterations looks like this 
```
I1120 12:25:54.422143     7 caffe.cpp:285] Starting Optimization
I1120 12:25:54.422150     7 solver.cpp:312] Solving RMNet_LReLU with splitted SSD head for PersonDetection
I1120 12:25:54.422152     7 solver.cpp:313] Learning Rate Policy: multistep
I1120 12:25:54.925384     7 solver.cpp:258] Iteration 0 (2.47475e-29 iter/s, 0.503185s/100 iters), loss = 1.90675
I1120 12:25:54.925411     7 solver.cpp:277]     Train net output #0: mbox_loss = 19.0675 (* 0.1 = 1.90675 loss)
I1120 12:25:54.925422     7 sgd_solver.cpp:174] Iteration 0, lr = 0.01
I1120 12:26:31.576131     7 solver.cpp:258] Iteration 100 (2.72856 iter/s, 36.6493s/100 iters), loss = 0.92784
.
.
.
I1121 03:45:27.105646     7 solver.cpp:258] Iteration 149900 (2.73074 iter/s, 36.6201s/100 iters), loss = 0.457963
I1121 03:45:27.105702     7 solver.cpp:277]     Train net output #0: mbox_loss = 3.26542 (* 0.1 = 0.326542 loss)
I1121 03:45:27.105708     7 sgd_solver.cpp:174] Iteration 149900, lr = 1e-05
I1121 03:46:03.390313     7 solver.cpp:614] Snapshotting to binary proto file snapshots/train_iter_150000.caffemodel
I1121 03:46:03.406065     7 sgd_solver.cpp:344] Snapshotting solver state to binary proto file snapshots/train_iter_150000.solverstate
I1121 03:46:03.545054     7 solver.cpp:350] Iteration 150000, loss = 0.458073
I1121 03:46:03.545073     7 solver.cpp:355] Optimization Done.
I1121 03:46:03.545075     7 caffe.cpp:296] Optimization Done.
``` 

But after fine-tuning the `get_detections.py` returns no detection or random boxes with conf = 0.2. 

Any idea what's getting wrong here? 

Results at confidence threshold 0.2 look like this. 

![frame_000001](https://user-images.githubusercontent.com/1974232/69516025-49f98880-0f72-11ea-9525-6d3e7ef04cc3.jpg)

**Comment by AlexanderDokuchaev** [comment]
hi @omair18 
Did you try get detection for `person_detection_0022.caffemodel`? Does it return same output?

**Comment by omair18** [comment]
Hello @AlexanderDokuchaev . 

Yes, ```person_detection_0022.caffemodel``` returns normal output. 

![frame_000001](https://user-images.githubusercontent.com/1974232/69537928-f2274580-0fa2-11ea-9349-84f9d8c8a208.jpg)

**Comment by AlexanderDokuchaev** [comment]
@omair18 
1. Check `DataLayer stats` output in log. It should contain names annotation files and number of annotated objects.

```
I1126 11:45:52.961249     6 _caffe.cpp:105] Found 1 tasks:
I1126 11:45:52.961262     6 _caffe.cpp:105]    0: /data/annotation/GM_D1_L1_BC.mp4.xml
I1126 11:45:52.961272     6 _caffe.cpp:105] Loading task /data/annotation/GM_D1_L1_BC.mp4.xml...
I1126 11:45:59.358994     6 _caffe.cpp:105] DataLayer stats: loaded 37763 frames with 417718 objects.
I1126 11:45:59.359105     6 _caffe.cpp:105] Labels:
I1126 11:45:59.359120     6 _caffe.cpp:105]    1: 417718 (100.00%) - 037763 frames
```

2. Check that after converting video to images, annotation are correct (some conversion tool can skip several frames in begin of video) .

3. Try to deacrease [base_lr](https://github.com/opencv/training_toolbox_caffe/blob/develop/models/person_detection/solver.prototxt#L6) and number of steps

```
max_iter: 50000

base_lr: 0.01
lr_policy: "multistep"
stepvalue: 30000
stepvalue: 40000
```

**Comment by omair18** [comment]
@AlexanderDokuchaev The ```DataLayer stats``` output looks like this 

```
I1120 12:25:52.786892     7 _caffe.cpp:105] DataLayer stats: loaded 7847 frames with 51082 objects.
I1120 12:25:52.786957     7 _caffe.cpp:105] Labels:
I1120 12:25:52.786969     7 _caffe.cpp:105]    1: 051082 (100.00%) - 007847 frames
I1120 12:25:52.813509     7 net.cpp:159] Setting up data
I1120 12:25:52.813547     7 net.cpp:166] Top shape: 4 3 400 680 (3264000)
I1120 12:25:52.813552     7 net.cpp:166] Top shape: 1 1 1 8 (8)
I1120 12:25:52.813555     7 net.cpp:174] Memory required for data: 13056032
I1120 12:25:52.813585     7 layer_factory.hpp:114] Creating layer data_data_0_split
I1120 12:25:52.813604     7 net.cpp:121] Creating Layer data_data_0_split

```
I have attached the log file and a sample input annotation file here. I will try the other recommendations that you have given. Thank you. :) 

[2019-11-20_17-25-48.log](https://github.com/opencv/training_toolbox_caffe/files/3893464/2019-11-20_17-25-48.log)
[annotation_file_6.xml.txt](https://github.com/opencv/training_toolbox_caffe/files/3893474/annotation_file_6.xml.txt)

**Comment by AlexanderDokuchaev** [comment]
@omair18 logs and xml looks ok. 
My be it's overfitting on your train subset. 
Try to set base_lr = 0.001 or 0.0001 and evaluate all snapshots since 1000 on validataion and train subsets.

**Comment by omair18** [comment]
@AlexanderDokuchaev 
I have retrained my dataset with base_lr=0.0001 and the issue still persists. Trained model till 1000 iterations doesn't show any detection (during evaluation) and for the rest of the snapshots the output remains the same as mentioned earlier.
 
Below are the results and log file of the last training. 
Log file:  [2019-12-04_11-12-09.log](https://github.com/opencv/training_toolbox_caffe/files/3920811/2019-12-04_11-12-09.log)


-iteration # 2000 
![frame_000001_2000](https://user-images.githubusercontent.com/1974232/70125266-52e80980-1698-11ea-8a05-5e53ff754862.png)

-iteration # 3000 
![frame_000001_3000](https://user-images.githubusercontent.com/1974232/70125284-5c717180-1698-11ea-99d5-a50a2a34260b.png)

- iteration # 10000 
![frame_000001_10000](https://user-images.githubusercontent.com/1974232/70125306-685d3380-1698-11ea-90eb-9eb45e2ab56a.png)

- iteration # 15000 
![frame_000001_15000](https://user-images.githubusercontent.com/1974232/70125327-74e18c00-1698-11ea-9d58-0e6dec84f540.png)

**Comment by AlexanderDokuchaev** [comment]
@omair18  in the last log, you don use trained snapshot to initialize weights.

In 2019-11-20_17-25-48.log was
```
I1120 12:25:54.405313     7 caffe.cpp:192] Finetuning from /init_weights/../init_weights/person_detection_0022.caffemodel
I1120 12:25:54.421008     7 upgrade_proto.cpp:114] Attempting to upgrade batch norm layers using deprecated params: /init_weights/../init_weights/person_detection_0022.caffemodel
```

**Comment by omair18** [comment]
@AlexanderDokuchaev yes in `2019-11-20_17-25-48.log` I wanted to see if the model trains something without the pretrained weights. However, I did another iteration with the pretrained weights and changed the training parameters. The results are still the same. I can't seem to figure out what is the issue here. 

Log File: [2019-12-09_10-51-02.log](https://github.com/opencv/training_toolbox_caffe/files/3943308/2019-12-09_10-51-02.log)


![frame_000004](https://user-images.githubusercontent.com/1974232/70501164-2839fd00-1b3f-11ea-8ab7-5c612885262b.png)

**Comment by AlexanderDokuchaev** [comment]
@omair18  
I'm really cant understand why it happens :disappointed: 

Could you provide you caffemodels and some train images with annotation?

**Comment by omair18** [comment]
@AlexanderDokuchaev 
Sure. I have shared the data folder on this link. 
https://mega.nz/#F!anYzwQbS!f_Czaym_kfbOK6_gvqe1JA

**Comment by omair18** [comment]
@AlexanderDokuchaev 
I tried to train the network on a very simple dataset i.e A single human placed randomly on the top of an empty image. But the results were still weird. I'm sharing the data and results here. 

![frame00000_1](https://user-images.githubusercontent.com/1974232/70560516-8fd96200-1baa-11ea-937a-6ff433dc66fd.png)

Results:
![frame0000_13](https://user-images.githubusercontent.com/1974232/70560676-d9c24800-1baa-11ea-8f86-b98df16269d5.png)

Data: https://mega.nz/#!ivQSXajY!NRwJUx8J0AbLW-N6JYze1EQBwIxpogADN2wBs5d6Sv0

**Comment by AlexanderDokuchaev** [comment]
@omair18 
Found two reasons:
1. Format of resolution in `train_tasks.txt` should be `height,width` instead of `width,height`
```
annotation/annotation_file_5.xml 1080,1920 images/video_5
```
2. Annotations looks weird. 
![capture]( https://user-images.githubusercontent.com/48012821/70566154-92d15400-1ba4-11ea-9c34-0eca36dbd761.jpg)

Check you annoation files by [test_data2.zip](https://github.com/opencv/training_toolbox_caffe/files/3947019/test_data2.zip), it's part of data_loader_layer with visualization.

```
python test_data2.py /mnt/datasets/data2/annotation/annotation_file_5.xml 1080,1920 /mnt/datasets/data2/images/video_5/
```

**Comment by omair18** [comment]
@AlexanderDokuchaev Thank you so much. Issue has been resolved. :)

**Comment by omair18** [comment]
@AlexanderDokuchaev  right now the person-detector is taking an input image dimension of ```3x400x680``` can I get the init weights for ```person-detection-retail-0013``` where input image dimensions are ```3x320x544``` ?

**Comment by AlexanderDokuchaev** [comment]
@omair18 
You can change input size in this files
https://github.com/opencv/training_toolbox_caffe/blob/develop/models/person_detection/data_config.json#L4
https://github.com/opencv/training_toolbox_caffe/blob/develop/models/person_detection/deploy.prototxt#L11
